add application monitoring for svc-bip-api

lisac opened the pull request:
## Summary 
Introduces `MetricLoggerService`, which exposes methods for submitting count and distribution metrics via a Datadog API client; and modifies `BipApiService` and a secondary error handling classes to invoke those metric submission methods.

The data points are submitted under these metric names: 
- vro_bip.request_start
- vro_bip.request_duration
- vro_bip.response_complete
- vro_bip.response_error
- vro_bip.listener_error
- vro_bip.message_conversion_error

Associated tickets or Slack threads:
- #2932 - contains screenshot of datadog dashboard with mock data

Datadog example code:
- [SubmitMetrics.java](https://github.com/DataDog/datadog-api-client-java/blob/master/examples/v1/metrics/SubmitMetrics.java)
- [SubmitDistributionPoints.java](https://github.com/DataDog/datadog-api-client-java/blob/master/examples/v1/metrics/SubmitDistributionPoints.java)

lisac committed:
introduce a service that submits metrics to datadog

lisac committed:
introduce a service that submits metrics to datadog

lisac committed:
fix syntax errors

lisac committed:
fix syntax errors

lisac committed:
fix format violations

lisac committed:
fix format violations

lisac committed:
fix format violations

lisac committed:
fix format violations

lisac committed:
add unit tests

lisac committed:
add unit tests

lisac committed:
remove nondraft condition on CI tests for BIP.

lisac committed:
remove nondraft condition on CI tests for BIP.

lisac committed:
fix payload - need to send along a timestamp

lisac committed:
fix payload - need to send along a timestamp

lisac committed:
fix (?) for spring config

lisac committed:
fix (?) for spring config

lisac committed:
fix tags submitted per event to use key:value format

lisac committed:
fix tags submitted per event to use key:value format

lisac committed:
Merge branch 'develop' into lisac/dd-svc-bip-api

lisac committed:
fix (?) for passing along the datadog values

lisac committed:
add metric logging to error handlers

lisac committed:
refactor configuration of datadog api client

lisac committed:
remove unneeded annotation

lisac committed:
merge

lisac committed:
Merge branch 'develop' into lisac/dd-svc-bip-api

lisac committed:
fix for lint/format

lisac committed:
fix (?) for spring boot error on integration tests: set a default value

lisac committed:
add a tag for the http method
to allow for querying in datadog

lisac committed:
remove unused file
(was renamed to DataClientConfig.java)

lisac committed:
rename metric  to be more descriptive

lisac committed:
use milliseconds instead of nanoseconds for request duration

lisac committed:
revert change to CI workflow 

(it was introduced in this PR, for convenience in testing the PR)

lisac committed:
update default value for DD_SITE
api.ddog-gov.com leads to an API client initialization error, whereas ddog-gov.com does not.

lisac committed:
Merge remote-tracking branch 'refs/remotes/origin/lisac/dd-svc-bip-api' into lisac/dd-svc-bip-api

msnwatson reviewed (approved):
Just noting we might want to externalize some of the code for our other Java services as we do future work 😀

lisac committed:
Merge branch 'develop' into lisac/dd-svc-bip-api

lisac committed:
Merge branch 'develop' into lisac/dd-svc-bip-api

lisac committed:
remove calls to MetricLoggerService from non-test code paths.
why: debugging why integration tests are failing. is there a bug in MetricLoggerService?

lisac committed:
Revert "remove calls to MetricLoggerService from non-test code paths."

This reverts commit 6609728bcefe2f806827cebf96e7d926f3f623ff.

lisac committed:
add more exception handling around MetricLoggerService
why: debugging failed integration tests on EP merge, seems caused by MetricLoggerService
